File: www/core/Polyfills/WCMessage/WCMessageChannel.js
importScripts("JSClassDelegate");
importScripts("WCMessagePort");

class WCMessageChannel extends JSClassDelegate {

    constructor(instance) {
        super();
        if (this.constructor.name !== "WCMessageChannel") { return; }
        this.instance = instance || scope && scope.createWCMessageChannel();
    }

    get port() { return WCMessagePort.create(this.instance.port); }

    static create(instance) {
        return new WCMessageChannel(instance);
    }

}

File: src/ios/core/APIs/WCMessage/WCMessageChannel.swift
//
//  WCMessageChannel.swift
//  WTVJavaScriptCore
//
//  Created by Jing KE on 29/06/16.
//  Copyright © 2016 WizTiVi. All rights reserved.
//

import Foundation
import JavaScriptCore

@objc protocol WCMessageChannelJSExport: JSExport {
    
    var port: WCMessagePort { get }
    
}

class WCMessageChannel: JSClassDelegate, WCMessageChannelJSExport {
    
    let port: WCMessagePort
    
    override init(context: ScriptContext) {
        self.port = WCMessagePort.create(context)
        super.init(context: context)
    }
    
}

extension WCMessageChannel {
    
    class func create(context: ScriptContext) -> WCMessageChannel {
        return WCMessageChannel(context: context)
    }
    
}